### ISSUE TITLE
Download the selected folders from the correct repo

### ISSUE THREAD

**Issue by c21sebar** [HGustavs/LenaSYS#13071]
In conjuction with #13069 and #13070 

Download the correct files and directories to the webserver if the files do not exist.
If the files exist on the webserver compare if changes have been made and download and replace if changes have been made, if not ignore.

**Comment by e21krida** [comment]
Research:
https://www.php.net/manual/en/function.file-put-contents.php - "file_put_contents" can be used to actually save a file locally, we can use the path information from the retrieved github repo to rebuild the file-structure locally.

https://www.php.net/manual/en/function.sha1-file.php - "sha1_file" can be used to generate a sha for a local file, if this is able to be used to compare with the sha for each individual file fetched from a github repository still remains to see.

https://www.php.net/manual/en/function.filesize.php - "filesize" can be used to check the size of a local file, this can otherwise be used to compare the file with the file fetched from the github repository if sha1_file doesn't work, chances of a file being the same size down to 1kb without differences is miniscule at best. 

https://www.php.net/manual/en/function.filemtime.php - "filemtime" can be used to check when a file was last updated locally, this could also be used to compare to the github repository in case sha1_file nor filesize doesn't work as intended in a comparator.

**Comment by e21krida** [comment]
https://www.php.net/manual/en/function.mkdir.php - "mkdir" can be used to create a local directory if one doesn't already exist, will be necessary in recreating the file-structure, "file_put_contents" doesn't seem to be able to create directory only files.

**Comment by e21krida** [comment]
@HGustavs @c21sebar We have successfully commited the code that manages to download and reconstruct the file-tree locally, however, it does not appear to present itself on the webserver. We don't receive an error message and after attempting to print_r the path the result is "/var/www/html/root/G3/students/a21olljo/LenaSYS/recursivetesting/Webbprogrammering-Examples/ajax_api_index.html". Is this a issue of not having the permissions to write the data? As this works flawlessly when tested locally, it wouldn't appear to be an issue regarding the actual code at hand.

**Comment by a21olljo** [comment]
Done research on how to utilize the existing system to download files found in fileed.php https://github.com/HGustavs/LenaSYS/wiki/Database-file-upload. This is yet not fully understood how it takes the file uploaded and extracts the information, to later put it into the databases table `filelink`. As well as how the selected file is uploaded to the directories under courses. 
What I want to do is to utilize this system of uploading files and do it for the download_url found in the metadata in github.

**Comment by a21olljo** [comment]
I have reached a point in this issue where I no longer know what to try, my understanding of the code gives me. 
What to do seems easy, where the goal is to take the files which is the output of traversing the repo (already done). Then take this file and feed it into the input found in fileed.php, which already takes an array of files. 
`<input name="uploadedfile[]" id="uploadedfile" type="file" multiple="multiple" placeholder="hej.text"/>`

Then when the button a bit further down is pressed the files are submitted and (what I understand) past to filereceive.php, where they are accessed individually one by one through POST and the `$_FILES['uploadedfile']` command. (but this part does not affect the issue following that we are only concerned with using the input) 

TL;DR: What we want that I can not seem to do, is to automatically fill the input field, which is accessed by POST and $_FILES['uploadedfile'].

**Comment by e21krida** [comment]
@HGustavs We have solved the downloading of an entire repo, but it doesn't seem like we can write to the LenaSYS folder, we can only write it outside like this "https://cms.webug.se/root/G3/students/e21krida/Webbprogrammering-Examples/", do you have any idea as to why this may be? Is it simply an issue regarding rights to writing in the folder or is there something we've missed?
